compute/volumes_test.go

```diff
@@ -0,0 +1,451 @@
+//
+// Copyright (c) 2018, Joyent, Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+package compute_test
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/joyent/triton-go/compute"
+	"github.com/joyent/triton-go/testutils"
+	"github.com/pkg/errors"
+)
+
+const fakeVolumeID = "1fd4ecb9-7f66-cf31-9a8a-b661d3adebcf"
+
+func TestGetVolume(t *testing.T) {
+	computeClient := MockIdentityClient()
+
+	do := func(ctx context.Context, cc *compute.ComputeClient) (*compute.Volume, error) {
+		defer testutils.DeactivateClient()
+
+		volume, err := cc.Volumes().Get(ctx, &compute.GetVolumeInput{
+			ID: fakeVolumeID,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return volume, nil
+	}
+
+	t.Run("successful", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountURL, "volumes", fakeVolumeID), getVolumeSuccess)
+
+		resp, err := do(context.Background(), computeClient)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if resp == nil {
+			t.Fatalf("Expected an output but got nil")
+		}
+	})
+
+	t.Run("eof", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountURL, "volumes", fakeVolumeID), getVolumeEmpty)
+
+		_, err := do(context.Background(), computeClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "EOF") {
+			t.Errorf("expected error to contain EOF: found %s", err)
```

Comment:
In the future make a sweep for `%s` as the verb for error messages and swap to `%v` that way we don't cargo cult `%s`.